tests/TestConditionalRotation.py

```
@@ -0,0 +1,190 @@
+    def test_normalize_structured_action_dropped(self):
+        ast, w = normalize_ast([{"t": "dodge"}, "1"])
+        self.assertEqual(ast, ["1"])
+        self.assertEqual(len(w), 1)
+
+    def test_normalize_invalid_sub_cond_in_all_dropped(self):
+        ast, w = normalize_ast([{"if": {"all": ["ult1", "bad_atom"]}, "then": ["1"]}])
+        self.assertEqual(ast, [{"if": {"all": ["ult1"]}, "then": ["1"]}])
+        self.assertTrue(any("非法条件原子" in x for x in w))
+
+    def test_normalize_all_fully_invalid_dropped(self):
+        ast, w = normalize_ast([{"if": {"all": ["bad1", "bad2"]}, "then": ["1"]}])
```

Ruff (0.16.0)

[warning] Unpacked variable `w` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)